### ISSUE TITLE
[fluent-langneg] Support BCP47 extended language subtags

### ISSUE THREAD

**Issue by tusooa** [projectfluent/fluent.js#551]
Currently, https://github.com/projectfluent/fluent.js/blob/8991a1760a83f1f9d8900e325ecc83dcb636337c/fluent-langneg/src/locale.ts only allow the primary language subtag, and [extended language subtag](https://tools.ietf.org/html/bcp47#section-2.2.2) is not implemented.

It also does not take into account of macrolanguages. For example, requesting `zh` when there is `cmn-Hans` available will not match `cmn-Hans`.

**Comment by zbraniecki** [comment]
Yes. We're planning to swap this for ICU4X once ICU4X matures a bit, and that will bring support for Unicode Extensions including extended langauge subtag.

**Comment by tusooa** [comment]
It's more than a year. Do we have any updates on this?

**Comment by eddieantonio** [comment]
Bumpin'.

Was trying this code, and it's totally broken:

```js
negotiateLanguages(['es-MX'], ['es', 'en'], { strategy: "matching" })
// expected: ['es']
// got: ['es'] 👍🏼

negotiateLanguages(['es-419'], ['es', 'en'], { strategy: "matching" }));
// expected: ['es']
// got: [] 👿
```
